### PR TITLE
Fix dry-run not stopping

### PR DESCRIPTION
**Pull request by charliedrage**
Fixes issue #339.

Dry-run isn't passed to stop

Found fix via `python -m trace -t` and some logging

### CODE REVIEW

File: atomicapp/nulecule/base.py [hunk 1]
```diff
@@ -192,21 +192,20 @@ def load(self, nodeps=False, dryrun=False):
...
         if self._app:
-            self._app.run(provider_key, dry)
+            self._app.run(provider_key, dryrun)
             return
-        provider_key, provider = self.get_provider(provider_key, dry)
+        provider_key, provider = self.get_provider(provider_key, dryrun)
         provider.artifacts = self.rendered_artifacts.get(provider_key, [])
         provider.init()
-        if not dry:
-            provider.deploy()
```
Comment (charliedrage) [inline atomicapp/nulecule/base.py]: @rtnpro 
Just to make sure we are on the same page, dryrun is handled from __within__ the provider, even though we pass dryrun, we should still be executing deploy... / undeploy

File: atomicapp/nulecule/base.py [hunk 2]
```diff
@@ -192,21 +192,20 @@ def load(self, nodeps=False, dryrun=False):
...
     def stop(self, provider_key=None, dryrun=False):
         if self._app:
-            self._app.stop(provider_key)
+            self._app.stop(provider_key, dryrun)
             return
-        provider_key, provider = self.get_provider(provider_key)
+        provider_key, provider = self.get_provider(provider_key, dryrun)
         provider.artifacts = self.rendered_artifacts.get(provider_key, [])
         provider.init()
         provider.undeploy()
```
Comment (charliedrage) [inline atomicapp/nulecule/base.py]: this is :+1: but we didn't pass dryrun at all in the first place :) it's all passing now.

File: atomicapp/nulecule/base.py (hunk 1 above)
Comment (rtnpro) [inline atomicapp/nulecule/base.py]: Agreed!

File: atomicapp/nulecule/base.py (hunk 2 above)
Comment (rtnpro) [inline atomicapp/nulecule/base.py]: Awesome :)

File: atomicapp/nulecule/base.py [hunk 3]
```diff
@@ -192,21 +192,20 @@ def load(self, nodeps=False, dryrun=False):
             else:
                 self.load_external_application(dryrun)
 
-    def run(self, provider_key, dry=False):
+    def run(self, provider_key, dryrun=False):
```
Comment (rtnpro) [inline atomicapp/nulecule/base.py]: :+1:

**Comment by rtnpro** [comment]
:+1: LGTM

**Comment by dustymabe** [comment]
LGTM

**Pull request merged**